### DISCUSSION TITLE
Add Next support

### DISCUSSION THREAD

**Discussion by thefinnomenon** [mmazzarolo/react-native-universal-monorepo#14] [Ideas]
I have been trying to add a package for Next.js but am running into two issues. 

1. The Next app starts and runs fine except it doesn't show the image from the app.
![image](https://user-images.githubusercontent.com/3837369/134783847-e1610192-ed59-484d-92c8-fe5e0a1bbd70.png)

2. At some point when I setup the Next package, it breaks the other web based packages; web, browser-ext, and electron. When I try to run them I get the following error,
```
Failed to compile.

../core/src/App.tsx
Module not found: Can't resolve 'react-native' in '<path>/<to>/packages/core/src'
```

There were some other annoying things like Next seemingly refusing to inject _DEV_ and a weird error about hook usage but I came up with solutions for those.

Sorry if you aren't interested in using Next but figured I would see if you have any insight -- this monorepo stuff is so confusing 😖.

Here is the working branch https://github.com/thefinnomenon/universal-project-template/tree/feat-next

**Comment by mmazzarolo** [comment]
@thefinnomenon I transformed the issue into a discussion 👍 
Can't work on it now, but I do have plans to give Next.js a try once I'm done with the blog posts (soon ™️ ). 

A couple of feedbacks:
- You shouldn't need [`react-native`](https://github.com/thefinnomenon/universal-project-template/blob/7cb7f25007ddf7af8b1c76eaafeb426f69a8aea1/packages/next/package.json#L18) as a dependency in the `next` workspace
- I recently did a huge refactoring of this repo that you might wanna integrate, see #10 
- "a weird error about hook usage" is probably related to multiple copies of React/React-DOM bundled within a single chunk. It should be solved by using aliases (like you're doing [here](https://github.com/thefinnomenon/universal-project-template/blob/7cb7f25007ddf7af8b1c76eaafeb426f69a8aea1/packages/next/next.config.js#L44) or like we're now doing in this repo using the `react-native-monorepo-tools` package)
- "web, browser-ext, and electron. FYI, when I try to run them I get the following error" when this happens, is often because of some misconfiguration in the nohoisting. Using the new refactored code with the `react-native-monorepo-tools` package might help here 👍

**Comment by Fiangor** [comment]
Sir, I am new here, my question is, how do I with your assistance add up or to connect my Trust Wallet Account with Wallet Connect. Please your Advice. Thanks

**Reply by mmazzarolo** [reply]
Pretty sure you're in the wrong thread :)

**Comment by mmazzarolo** [comment]
#16 Next.js support is now available in the repo!

**Reply by liamrohan** [reply]
So glad to have found your repo, thanks for working on this.
